scripts/remove_unused.py

```diff
@@ -34,40 +38,60 @@
 while lines_removed_this_time > 0:
     lines_removed_this_time = 0
     for path in zig_files:
+        # get all lines of code
         with open(path) as f:
             orig_file = f.read()
         orig_lines = orig_file.split("\n")
         if orig_lines[-1] == "":
             orig_lines = orig_lines[0:-1]
+
+        # identify imports
         imported_names = []
         for line_num, line in enumerate(orig_lines):
             match = import_line_regex.match(line)
             if match:
                 imported_names.append((match.groups()[0], line_num))
         lines_to_drop = set()
         num_lines_to_remove = 0
+
+        # identify which imports are unused
         for name, line in imported_names:
             match = re.findall(f"[^a-zA-Z0-9_.]{name}[^a-zA-Z0-9_]", orig_file)
             assert len(match) > 0
             if len(match) == 1:
                 lines_to_drop.add(line)
                 num_lines_to_remove += 1
-        with open(path, "w") as f:
-            f.writelines(
-                f"{line}\n"
-                for i, line in enumerate(orig_lines)
-                if i not in lines_to_drop
-            )
-        lines_to_drop
-        print(path, num_lines_to_remove)
+
+        # do something about the unused imports
+        if num_lines_to_remove:
+            if args.check:
+                print(f"Found {num_lines_to_remove} unused import(s) in {path}:")
+                padding = int(math.log(max(lines_to_drop), 10))
+                for line_num in sorted(lines_to_drop):
+                    line_num_str = f"{line_num}".rjust(padding, " ")
+                    print(f"{line_num_str} | {orig_lines[line_num]}")
+                print()
+            else:
+                with open(path, "w") as f:
+                    f.writelines(
+                        f"{line}\n"
+                        for i, line in enumerate(orig_lines)
+                        if i not in lines_to_drop
+                    )
+                print(path, num_lines_to_remove)
+                os.system(f"zig fmt {path}")
+        elif args.verbose:
+            print(path, num_lines_to_remove)
+
         total_num_lines_removed += num_lines_to_remove
         lines_removed_this_time += num_lines_to_remove
-        if (num_lines_to_remove > 0):
-            os.system(f"zig fmt {path}")
 
-    print("removed this iteration:", lines_removed_this_time)
-    print()
+    if args.check:
+        break
+    else:
+        print("removed this iteration:", lines_removed_this_time)
+        print()
 
-print("total lines removed:", total_num_lines_removed)
-if (total_num_lines_removed > 0):
+print("total unused found:", total_num_lines_removed)
```

Comment:
If the --check flag is not passed, the output would be something like

```
src/trace/log.zig 1
removed this iteration: 1

removed this iteration: 0

total unused found: 1
```

Might be nice to have the message say Removed, instead of Found. And something about the files where the unused lines has been removed.

Something like:

`Removed x unused import(s) in src/trace/log.zig:`

Reply:
4d21abecd72dcdf48bbd79eb95680da047435f22